balrog/os-de-support.rst

```diff
@@ -0,0 +1,198 @@
+* Create a new mar with the script, something like:::
+
+    create-channel-switch-mar.py --version 115.0 --channel esr --accepted-channel-id firefox-mozilla-esr --mar-channel-id firefox-mozilla-release --distribution-id mozilla-mac-eol-esr115 --output switch-to-esr115.0-eol-mac.mar
+
+* Examine your mar. Note that mar contents are xz (or bzip2) compressed, even after extraction.::
+
+    mar -T switch-to-esr115.0-eol-mac.mar
+    mar -x switch-to-esr115.0-eol-mac.mar
+    xz -c -d updatev3.manifest
+
+* Sign the mar via adhoc_signing. See https://github.com/mozilla-releng/adhoc-signing/blob/main/signing-manifests/bug1835022.yml for a sample manifest. 
+
+Verify the channel-switching mar
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+TBD 
+
+Copy the channel-switching mar to archive.mozilla.org
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Channel-switching mar files are typically copied to https://archive.mozilla.org/pub/firefox/releases/custom-updates/. You may want to file a bug like `bug 1839195 <https://bugzilla.mozilla.org/show_bug.cgi?id=1839195>`__.
+
+Balrog updates
+~~~~~~~~~~~~~~
+
+Each Firefox channel likely needs a new rule in Balrog.
+
+Broadly there are 3 behaviors of interest:
+
+* ESR-SWITCH: switch de-supported users to the ESR channel using a channel-switching mar
+* DE-SUPPORT: allow updates to the most recent supported version on the channel, then serve a "de-support" update which points to the relevant KB article
+* NO-UPDATE: stop all updates temporarily
+
+Typically NO-UPDATE is used to stop updates temporarily just before the first unsupported version is available on each channel. Once the last supported version is available, an ESR-SWITCH or DE-SUPPORT rule is implemented and the NO-UPDATE rule is deleted.
+
+Update Balrog rules for Nightly
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For the Nightly channel, we usually DE-SUPPORT.
+
+Just *before* the first nightly build (could be a few days earlier, if convenient) of the first unsupported version, pause Nightly updates for the affected OS version: 
+
+* Create a new rule matching the ``Firefox : Nightly*`` channel and the OS version to serve the ``NoUpdate`` release.
+
+Shortly *after* the first nightly build of the first unsupported version:
+
+* Determine the build ID of the last build of the last supported version: You might check `archive.mozilla.org <https://archive.mozilla.org/pub/firefox/nightly/>`__ for instance.
+* Create a rule to pin Nightly* users to the most recent supported version: match the ``Firefox : Nightly*`` channel, the affected OS version and build ID < latest supported build ID determined above; serve the Release corresponding to that build ID.
+* Update the Release referenced by the new pinning rule, to include aliases (this is only required on Nightly, as it is caused by `bug 1810740 <https://bugzilla.mozilla.org/show_bug.cgi?id=1810740>`__): Update the Release blob by adding aliases like::
```

Comment:
Let's also explicitly says the up-to-date list of aliases should be looked up in Firefox-mozilla-central-nightly-latest, just in case they change before we have to do this again.